velox/exec/RowContainer.h

```diff
@@ -17,14 +17,52 @@
 
 #include "velox/common/memory/HashStringAllocator.h"
 #include "velox/common/memory/MemoryAllocator.h"
-#include "velox/exec/Aggregate.h"
+#include "velox/core/PlanNode.h"
 #include "velox/exec/ContainerRowSerde.h"
 #include "velox/exec/Spill.h"
 #include "velox/vector/FlatVector.h"
 #include "velox/vector/VectorTypeUtils.h"
 
 namespace facebook::velox::exec {
 
+class Aggregate;
+
+class Accumulator {
+ public:
+  Accumulator(
+      bool isFixedSize,
+      int32_t fixedSize,
+      bool usesExternalMemory,
+      int32_t alignment,
+      std::function<void(folly::Range<char**> groups)> destroyFunction);
+
+  Accumulator(Aggregate* aggregate);
+
+  bool isFixedSize() const;
+
+  int32_t fixedWidthSize() const;
+
+  bool usesExternalMemory() const;
+
+  int32_t alignment() const;
+
+  void destroy(folly::Range<char**> groups);
+
+  /// Used only for spilling. Do not introduce other usages.
+  Aggregate* aggregateForSpill() const {
```

Comment:
This seems defeating the purpose of this decoupling.  How hard is it to let `Spiller` keep both row container and aggregates directly?

Reply:
@Yuhta This might be possible, but the change is already quite large. If possible, I'd like to land this change and take up the spiller separately.

Reply:
@Yuhta This change is valuable by itself as it allows for much cleaner implementation of TopNRowNumber operator without creating a synthetic aggregate. See https://github.com/facebookincubator/velox/pull/5110

For the original version, search TopNAggregate in

https://github.com/facebookincubator/velox/compare/171f07c3e20a1482d2692531e254f34149058207..9b717e4e989becf007833e38dab0aaeb35387bf0

Reply:
Let's do it in a separate PR then